tests/acceptance/features/bootstrap/RoleAssignmentContext.php

```diff
@@ -0,0 +1,246 @@
+		return $roleToAssign["id"];
+	}
+
+	/**
+	 * @param string $user
+	 * @param string $userId
+	 * @param string $roleId
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function setRoleToUser($user, $userId, $roleId): void {
+		$this->sendRequestToAssignRoleToUser($user, $userId, $roleId);
+
+		if ($this->featureContext->getResponse()) {
+			$rawBody = $this->featureContext->getResponse()->getBody()->getContents();
+			if (isset(\json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR)["assignment"])) {
+				$assignment = \json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR)["assignment"];
+			}
+		}
+
+		Assert::assertEquals($userId, $assignment["accountUuid"]);
+		Assert::assertEquals($roleId, $assignment["roleId"]);
+	}
+
+	/**
+	 * @When /^user "([^"]*)" changes his own role to "([^"]*)"$/
+	 *
+	 * @param string $user
+	 * @param string $role
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userChangeOwnRole(string $user, string $role): void {
+		// we assume that the user knows uuid role.
+		$roleId = $this->userGetRoleIdByRoleName($this->featureContext->getAdminUserName(), $role);
+		$userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id');
+		$this->sendRequestToAssignRoleToUser($user, $userId, $roleId);
+	}
+
+	/**
+	 * @When /^user "([^"]*)" changes the role "([^"]*)" for user "([^"]*)"$/
+	 *
+	 * @param string $user
+	 * @param string $role
+	 * @param string $assignedUser
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userChangeRoleAnotherUser(string $user, string $role, string $assignedUser): void {
+		// we assume that the user knows uuid role.
+		$roleId = $this->userGetRoleIdByRoleName($this->featureContext->getAdminUserName(), $role);
+		$userId = $this->featureContext->getAttributeOfCreatedUser($assignedUser, 'id');
+		$this->sendRequestToAssignRoleToUser($user, $userId, $roleId);
+	}
+
+	/**
+	 * @When /^user "([^"]*)" tries to get assignments list$/
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function userGetAssignmentsList(string $user): void {
+		$userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id');
+		$this->sendRequestAssignmentsList($user, $userId);
+	}
+
+	/**
+	 * @When /^user "([^"]*)" should have the role "([^"]*)"$/
+	 *
+	 * @param string $user
+	 * @param string $role
+	 *
+	 * @return void
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function userShouldHaveRole(string $user, string $role): void {
+		$userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id');
+		$this->sendRequestAssignmentsList($this->featureContext->getAdminUserName(), $userId);
+		$rawBody =  $this->featureContext->getResponse()->getBody()->getContents();
+		if (isset(\json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR)["assignments"])) {
```

Comment:
this is dangerous, is the response does not have any "assignments" this test will silently pass as the assertion will not run

Reply:
Yes, true. I changed it. thanks